Course/Course/Presentation/Dates/CourseDatesView.swift

```diff
@@ -68,6 +89,29 @@ public struct CourseDatesView: View {
         )
         .frame(maxWidth: .infinity, maxHeight: .infinity)
     }
+    
+    private func showDatesSuccessView(title: String, message: String) -> some View {
+        if viewModel.eventState == .shiftedDueDates {
+            return DatesSuccessView(
+                title: title,
+                message: message,
+                selectedTab: .dates,
+                courseDatesViewModel: viewModel
+            )
+        } else {
+            return DatesSuccessView(
+                title: title,
+                message: message,
+                selectedTab: .dates
+            ) {
+                Task {
+                    await MainActor.run {
+                        viewModel.eventState = CourseDatesViewModel.EventState.none
+                    }
+                }
+            }
```

Comment:
Can it be simplified just to something like this?
```
return DatesSuccessView(
                title: title,
                message: message,
                selectedTab: .dates
            ) {
                viewModel.eventState = CourseDatesViewModel.EventState.none
            }
```

Reply:
This seems a bit odd. We switch to a `Task` only to switch back to the MainActor.

Reply:
@volodymyr-chekyrta Thank you for bringing that to my attention. I've streamlined the logic for simplification.

**Explanation:**
The `DatesSuccessView` `struct` was previously invoking action closures asynchronously on a background thread. Upon receiving the closure action, we had to switch back to the main thread to avoid warnings such as:

```Publishing changes from background threads is not allowed; make sure to publish values from the main thread (via operators like receive(on:)) on model updates.```

I've refactored this logic so that the action closures in `DatesSuccessView` are no longer asynchronous.